snooty/postprocess.py

```diff
@@ -1568,14 +1568,27 @@ def __init__(self, context: Context) -> None:
         self.pending_targets: List[n.Node] = []
 
     def enter_node(self, fileid_stack: FileIdStack, node: n.Node) -> None:
-        if not isinstance(node, (n.Target, n.Section, n.TargetIdentifier)):
+        if not isinstance(node, (n.Target, n.Section, n.TargetIdentifier)) and not (
+            isinstance(node, n.Directive) and "heading" in node.options.keys()
```

Comment:
You don't need to specify the `keys()` method

Reply:
removed!